src/HomeWork_2/routers/Groups/create-group.ts

```diff
@@ -1,12 +1,14 @@
-import { Request, Response } from 'express';
+import { NextFunction, Request, Response } from 'express';
 import { createGroupService } from '../../services/Groups/create-group';
+import { logRequestData } from '../../helpers/log-request-data';
 
-export const createUserHandler = async ({ body }: Request, res: Response) => {
+export const createUserHandler = async (req: Request, res: Response, next: NextFunction) => {
   // TODO - add parameters validation
-  const result = await createGroupService(body.name, JSON.parse(body.permissions));
-  if (result) {
+  try {
+    await createGroupService(req.body.name, JSON.parse(req.body.permissions));
     res.status(200).end();
-  } else {
-    res.status(500).end();
+  } catch (err) {
+    logRequestData(req, true);
```

Comment:
Not Node.js specific note, but in general, it's better to write a separate function instead of using boolean flags. If it's hard to avoid using the flags, at least we could create a wrapper function (e.g. `logFailedRequestData(req)`) that would incapsulate the "magic" flags, so we don't have to guess what `true` means in this context.